### ISSUE TITLE
We can claim before fake verification is finished

### ISSUE THREAD

**Issue by Th0rgal** [starknet-id/starknet.quest#574]
<img width="823" alt="image" src="https://github.com/starknet-id/starknet.quest/assets/41830259/90467392-701e-4c42-8c57-5221e793f4a8">

I clicked on verifying a twitter interaction.

**Comment by Dharmik79** [comment]
@ayushtom  I would like to work on this

**Comment by ayushtom** [comment]
@Dharmik79 assigned to you!

**Comment by Dharmik79** [comment]
@ayushtom  Can you add some more details for this issue, like how to reproduce it?

**Comment by ayushtom** [comment]
You need to open a quest on which there is twitter task.
complete all the task except the twitter task.
Now at the end you click on verify on the twitter task and you will observe that the `Get Reward` button is also enabled while the loader is running for this task at the same time. We would need to enable this button only after the loader is finished and the task is verified

**Comment by ayushtom** [comment]
@Dharmik79 what's the status on this?

**Comment by Dharmik79** [comment]
@ayushtom I am not able to see any quests on the screen. Am I missing something? I have cloned the env example file, but was not getting any quests on the dashboard.

**Comment by ayushtom** [comment]
can you check if you are using this as the api endpoint - sepolia.api.starknet.quest
Also this issue shouldn't take so long so you can do it quickly and if you have any questions, feel free to ask in the telegram group we have. You dont have to wait for me to ask 😉

**Comment by Dharmik79** [comment]
@ayushtom I am not able to replicate the issue, I have tried for different quests. 

However, I have found an issue regarding the domain registration link. In the Sepolia testnet, when attempting to register a domain, the link redirects to either the mainnet or the Goerli testnet instead of remaining within the Sepolia testnet environment.

**Comment by ayushtom** [comment]
hey @Dharmik79 , can you try doing the carbonable quest on sepolia and see if you can reproduce the issue?  Domain regiatration issue can be handled in a seperate issue and is not part of the scope. I would suggest trying to fix this first

**Comment by ayushtom** [comment]
you can also try for other quests on mainnet since this must be there too.

**Comment by Dharmik79** [comment]
Ok I will check it

**Comment by ayushtom** [comment]
@Dharmik79 are you able to work on the issue now?

**Comment by Dharmik79** [comment]
@ayushtom I was not able to find the issue, also after resetting the code to the default, I was not able to see any quests, even after setting the endpoint as **sepolia.api.starknet.quest**. I would not be able to find the issue. Please assign this to other dev.

Thank you for helping @ayushtom. Also, can you create the documentation for setting the application for new users, as the wallets are not connecting properly?

**Comment by ayushtom** [comment]
@Dharmik79 sure. Can you please elaborate on the exact issues you faced and i can add it to the readme docs. something like an FAQ. I have added base instructions to get the project starting but i would add these details too for getting started with setting up the starknet tooling on the system.

**Comment by Dharmik79** [comment]
@ayushtom  Below, are the issues I faced when setting up and running the code.

Firstly, I was checking for the documentation to set up the code on my machine. It should state which version of node is used for the installation. Also, the script should be a one-liner, to install all the necessary packages.
Secondly, I was not sure which wallet to use, I have used all the listed wallets, sometimes it works well, and sometimes it does not work. 
For bravos, sometimes it list all the quests and other times it keeps loading
For Argent, I was not getting any quests when connecting the argent web version
For email, I was not able to see the quests as well.

**Comment by ayushtom** [comment]
`Also, the script should be a one-liner, to install all the necessary packages.`

This should be `npm i` right? what else was missing here?


```
Secondly, I was not sure which wallet to use, I have used all the listed wallets, sometimes it works well, and sometimes it does not work.
For bravos, sometimes it list all the quests and other times it keeps loading
For Argent, I was not getting any quests when connecting the argent web version
For email, I was not able to see the quests as well.
```

Can you please share the network requests response you get for `/get_quests`

**Comment by Oshioke-Salaki** [comment]
Please I would love to be assigned this

**Comment by ayushtom** [comment]
@Oshioke-Salaki assigned!
